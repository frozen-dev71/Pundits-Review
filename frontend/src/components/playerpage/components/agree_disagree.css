/*
// ======================================
// CSS for agree / disagree buttons and container
// ======================================
*/


.agree-disagree-wrapper {
	padding: 2.5em;
    display: flex;
    width: 50vw;
    justify-content: space-around;

}

.agree-disagree-button-container {
	display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 15vh;
    align-items: center;
}

.agree-disagree-heading-text {
	letter-spacing: .1em;
	font-size: .75em;
}

.agree-disagree-button {
	background-color: #a2aab0;
    height: 3em;
    border-radius: 2em;
    cursor: pointer;
    width: 3em;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    border: none;
    opacity: .5;
    box-shadow: black 2px 2px 5px;
    transition: transform 0.25s ease,
                opacity 0.25s ease;
}

.agree-disagree-icon {
	font-size: 2em;
	color: white;
}

.agree-disagree-database-score {
	font-size: .75em;
}

.agree-disagree-button:active {
    transform: scale(.80);
    opacity: .4;
}

.agree-disagree-button:disabled {
    transform: none;
    box-shadow: grey 2px 2px 5px;
    background-color: #cccccc;
    color: #666666;
}

/*
// ======================================
// MEDIA QUERIES
// ======================================
*/


@media (min-width: 800px) {

    .agree-disagree-wrapper {

        width: 30vw;
    }
}


@media (min-width: 1200px) {

    .agree-disagree-wrapper {

        width: 20vw;
    }
}