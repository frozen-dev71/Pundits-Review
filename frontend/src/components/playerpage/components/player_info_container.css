/*
// ======================================
// CSS for the Player Info container on the PLAYER PAGE
// ======================================
*/


.player-info-container-wrapper {
	display: grid;
    grid-template-columns: repeat(6,1fr);
    grid-template-rows: repeat(5,1fr);
    grid-gap: .5em;
    width: 90vw;
    padding: 2em 0;
    align-items: center;
    margin-top: 7.5vh;
}

.player-page-player-shirt {
	grid-row: 2/6;
    grid-column: 1/3;
    width: 8em;
    height: 9em;
    align-self: center;
}

.player-page-player-info-name {
	grid-row: 1/3;
    grid-column: 3/7;
    font-size: 1.25em;
}

.player-page-player-info-position {
	grid-row: 3;
	grid-column: 4/7;
	font-size: 1em;
	margin-left: 1em;
}

.player-page-player-info-club-text {
	grid-row: 4;
	grid-column: 4/7;
	font-size: 1em;
	color: black;
	margin-left: 1em;
}

.player-page-player-info-nationality-text {
	grid-row: 5;
	grid-column: 4/7;
	font-size: 1em;
	margin-left: 1em;
}


.flag-icon-player-page-info-container {
	font-size: 1.25em;
    grid-row: 5;
    grid-column: 3;
    justify-self: center;
}

.club-badge-player-page-info-container {
	width: 2em;
    height: 2em;
    grid-row: 4;
    grid-column: 3;
}


/*
// ======================================
// MEDIA QUERIES
// ======================================
*/



@media (min-width: 600px) {

	.player-info-container-wrapper {
		width: 75vw;
	}

	.player-page-player-info-name {
		font-size: 1.5em;
	}

	.player-page-player-shirt {
		width: 10em;
    	height: 11em;
	}
}


@media (min-width: 800px) {

	.player-info-container-wrapper {
		width: 55vw !important;
		padding: 3em 0;
		grid-gap: 1em;
	}


	.player-page-player-shirt {
		width: 12em;
    	height: 13em;
	}
}


@media (min-width: 1000px) {


	.player-page-player-info-name {
		font-size: 2em;
	}


	.player-page-player-info-club-text, .player-page-player-info-nationality-text, .player-page-player-info-position {
		font-size: 1.25em;
	}

	.flag-icon-player-page-info-container {
		font-size: 1.5em;
	}


	.club-badge-player-page-info-container {
		width: 3em;
	    height: 3em;
	}

}

@media (min-width: 1400px) {

	.player-info-container-wrapper {
		width: 35vw !important;
		padding: 4em 0;
		grid-gap: 1.5em;
	}

}
