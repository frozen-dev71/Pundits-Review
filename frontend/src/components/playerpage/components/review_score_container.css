/*
// ======================================
// CSS for the Review Score Container used in both the players page and clubs page
// ======================================
*/


.review-score-container-wrapper {

	display: flex;
	flex-direction: column;
	justify-content: space-around;
	padding: 1em 0;
	align-items: center;
	height: 35vh;
}


.review-score-pundits-title {
	font-size: 1.25em;
}

.review-score-positive-negative-total-grid {
	display: grid;
    grid-template-columns: repeat(6,1fr);
    grid-template-rows: repeat(3,1fr);
    align-items: center;
    grid-gap: .5em;
    width: 50vw;
    font-size: .85em;
}


.review-score-form-text {
	font-size: 1.15em;
}

.review-score-positive-text {
	grid-row: 1;
	grid-column: 3/7;
}

.review-score-negative-text {
	grid-row: 2;
	grid-column: 3/7;
}

.review-score-total-text {
	grid-row: 3;
	grid-column: 3/7;
}

.review-score-positive-number {
	grid-row: 1;
	grid-column: 1;
	justify-self: right;
}

.review-score-negative-number {
	grid-row: 2;
	grid-column: 1;
	justify-self: right;
}

.review-score-total-number {
	grid-row: 3;
	grid-column: 1;
	justify-self: right;
}


@media (min-width: 600px) {

	.review-score-positive-negative-total-grid {

		width: 40vw;
	}
}


@media (min-width: 800px) {

	.review-score-positive-negative-total-grid {

		width: 25vw;
	}

	.review-score-pundits-title {
		font-size: 1.5em;
	}
}

@media (min-width: 1000px) {

	.review-score-positive-negative-total-grid {

		width: 25vw;
    	font-size: 1.25em;
	}

	.review-score-pundits-title {
		font-size: 1.75em;
	}


	.review-score-container-wrapper {
		height: 40vh;
	}

}


@media (min-width: 1400px) {

	.review-score-positive-negative-total-grid {

		width: 20vw;
	}

}


@media (min-width: 1600px) {

	.review-score-positive-negative-total-grid {

		width: 15vw;
	}

}

