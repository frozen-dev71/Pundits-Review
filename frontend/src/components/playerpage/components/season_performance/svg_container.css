/*
// ======================================
// CSS for Season Performance Graph
// ======================================
*/



#performance-graph-svg-container {

	width: 100%;
  height: 100%;
  overflow: visible;
  font-size: .5em;
  letter-spacing: .15em;
  background-image: url("https://i.ibb.co/7jKPG25/graph-background.png");
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
}


.grid line {
  stroke: lightgrey;
  stroke-opacity: 0.7;
  shape-rendering: crispEdges;
}

.grid path {
  stroke-width: 0;
}

.tick line {
	stroke-opacity: 0;
}

.grid .tick line {
	stroke-opacity: 0.4;
}

.domain {
	stroke: lightgrey;
  	stroke-opacity: 0.9;
  	shape-rendering: crispEdges;
}




/*
// ======================================
// MEDIA QUERIES
// ======================================
*/


@media (min-width: 650px) {

  #performance-graph-svg-container {

    font-size: .65em;

  }

}


@media (min-width: 1200px) {

  #performance-graph-svg-container {

    font-size: .75em;

  }

}
