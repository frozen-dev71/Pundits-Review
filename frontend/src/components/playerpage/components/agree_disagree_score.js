// ======================================
// Agree / Disagree Score Component - Diplays the score number for a player / club & is updated with put request response
// ======================================


// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { connect } from "react-redux";
import { usePromiseTracker } from "react-promise-tracker";

// Local JS Imports


// CSS Imports 
import './agree_disagree.css';


// ===========
// Agree / Disagree Score Component
// ===========


export class AgreeDisagreeScoreWrapper extends Component {


	render() {

		let arrayChoice;

		if (this.props.clubOrPlayer === 'club') {
			arrayChoice = this.props.clubAgreeDisagreeScores;
		} 

		if (this.props.clubOrPlayer === 'player') {
			arrayChoice = this.props.agreeDisagreeScores;
		} 

		let result;

		if (arrayChoice.length) {

			result = (

				<p className="agree-disagree-database-score body-text-light">{arrayChoice[arrayChoice.length -1][this.props.choice]}</p>

			);

		} else {
	

			result = (

				<span></span>

			);


		}


		return result;
	
	}

	
}




// ===============
// Connect redux state container to PlayerPage component
// ===============


const mapStateToProps = state => {
  return { 
  		
	  	agreeDisagreeScores: state.agreeDisagreeScores,
	  	clubAgreeDisagreeScores: state.clubAgreeDisagreeScores
  	};
};


export const AgreeDisagreeScore = connect(mapStateToProps)(AgreeDisagreeScoreWrapper);