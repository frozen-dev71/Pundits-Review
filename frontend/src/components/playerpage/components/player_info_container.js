// ======================================
// Player Info Container Component for use within the player page - name, image, position, club and country
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { Link } from 'react-router-dom';
import { connect } from "react-redux";

// Local JS Imports
import { country_codes } from '../../global/resources/ISO_COUNTRY_CODES';
import { flagFinder } from "../../global/functions/set_data_helpers";
import { setClub } from '../../actions/actions';
import { club_shirts } from '../../global/resources/club_shirts';

// CSS Import
import './player_info_container.css';



// ===========
// Player Info Container
// ===========

export class PlayerInfo extends Component {

	// Set the club - on link click
	_handleClubClick(clubName) {

		this.props.setClub(this.props.playerData.club_name);
	}


	render() {

		// Set specific club URL
		let clubNameParam = "/clubs/" + encodeURIComponent(this.props.playerData.club_name);

		let result = (

			<div className="player-info-container-wrapper">

				<span className={"club-shirt player-page-player-shirt " + club_shirts[this.props.playerData.club_name]} ></span>
				<p className="player-page-player-info-name heading-text" >{this.props.playerData.player}</p>
				<p className="player-page-player-info-position body-text-light" >{this.props.playerData.position}</p>
				<Link to={clubNameParam} className="player-page-player-info-club-text body-text-light react-link" onClick={() => this._handleClubClick()}>{this.props.playerData.club_name}</Link>
				<p className="player-page-player-info-nationality-text body-text-light" >{this.props.playerData.nationality}</p>

				<span className={this.props.playerNationalityClassName}></span>
				<Link to={clubNameParam} onClick={() => this._handleClubClick()} className={this.props.playerClubBadgeClassName}></Link>

			</div>

			);

		return result;
	}
}


// ===============
// Connect redux state container to Player Info component
// ===============


const mapStateToProps = state => {
  return { 

	  	clubsViewed: state.clubsViewed
  	};
};



// Export connected component
export const PlayerInfoContainer = connect(mapStateToProps, { setClub })(PlayerInfo);
