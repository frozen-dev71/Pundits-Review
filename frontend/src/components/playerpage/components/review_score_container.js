// ======================================
// Review Score Container Component for use within the player page - review score, positive, negative, total
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";

// Local JS Imports
import { ReviewScoreCircle } from '../../results/components/review_score_circle';

// CSS Imports 
import './review_score_container.css';

// ===========
// Review Score Component
// ===========


export class ReviewScoreContainer extends Component {


	render() {


		let result = (

			<div className="review-score-container-wrapper">

				<p className="review-score-pundits-title heading-text">Pundits Review</p>
				<ReviewScoreCircle reviewScore={this.props.reviewScore} />

				<div className="review-score-positive-negative-total-grid">

					<p className="review-score-positive-text body-text-light">Positive Reviews</p>
					<p className="review-score-negative-text body-text-light">Negative Reviews</p>
					<p className="review-score-total-text body-text-light">Total Reviews</p>

					<p className="review-score-positive-number body-text-light">{this.props.positiveReviews}</p>
					<p className="review-score-negative-number body-text-light">{this.props.negativeReviews}</p>
					<p className="review-score-total-number body-text-light">{this.props.totalReviews}</p>

				</div>

				
			</div>


			);

		return result
	}
}