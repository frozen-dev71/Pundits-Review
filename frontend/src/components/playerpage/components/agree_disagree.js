// ======================================
// PLAYERS Agree / Disagree Button Component allowing users to interact with player / club pages depending on their opinion
// Will post to the DB with number of agrees / disagrees
// ======================================


// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import axios from 'axios';
import { connect } from "react-redux";


// Local JS Imports 
import { updatePlayersAgreeDisagreeScore, setPlayersAgreeDisagreeScore, locallyUpdateAgreeDisagreeScore } from "../../actions/actions";
import { AgreeDisagreeScore } from './agree_disagree_score';

// CSS Imports 
import './agree_disagree.css';

// ===========
// Agree Disagree Component
// ===========

export class AgreeDisagreeContainer extends Component {

	constructor(props) {
		super(props);
		this.state = {
			
			buttonDisabled: false
		}
		this._handleClickAgree = this._handleClickAgree.bind(this);
		this._handleClickDisagree = this._handleClickDisagree.bind(this);
	}

	componentDidMount() {

		this.props.setPlayersAgreeDisagreeScore(this.props.playerName)

	}

	_handleClickAgree(e) {

		let currentPlayer = this.props.playersViewed[this.props.playersViewed.length - 1];

		updatePlayersAgreeDisagreeScore(currentPlayer, true);

		let newScoreObject = this.props.agreeDisagreeScores[this.props.agreeDisagreeScores.length -1];

		newScoreObject.agree_score += 1;

		this.props.locallyUpdateAgreeDisagreeScore(newScoreObject);

		// // Change Presentation of button
		// const currentScore = parseInt(document.querySelector('#agree-score').innerHTML)
		// const button = document.querySelector('#agree-score')
		// button.innerHTML = currentScore + 1

		// Disable the button
		this.setState({
			buttonDisabled: true
		})

	}

	_handleClickDisagree(e) {

		let currentPlayer = this.props.playersViewed[this.props.playersViewed.length - 1];

		updatePlayersAgreeDisagreeScore(currentPlayer, false);

		let newScoreObject = this.props.agreeDisagreeScores[this.props.agreeDisagreeScores.length -1];

		newScoreObject.disagree_score += 1;

		this.props.locallyUpdateAgreeDisagreeScore(newScoreObject);

		// // Change Presentaton of button
		// const currentScore = parseInt(document.querySelector('#disagree-score').innerHTML)
		// const button = document.querySelector('#disagree-score')
		// button.innerHTML = currentScore + 1

		// Disable the button
		this.setState({
			buttonDisabled: true
		})

	}

	render() {


		let result = (

			<div className="agree-disagree-wrapper">

				<div className="agree-disagree-button-container">

					<p className="agree-disagree-heading-text body-text-light">AGREE</p>
					<button id="agree-button" className="agree-disagree-button" disabled={this.state.buttonDisabled} onClick={(e) => this._handleClickAgree(e)}><span className="agree-disagree-icon fa fa-check"></span></button>
					<AgreeDisagreeScore choice={"agree_score"} clubOrPlayer={"player"}/>

				</div>

				<div className="agree-disagree-button-container">

					<p className="agree-disagree-heading-text body-text-light">DISAGREE</p>
					<button id="disagree-button" className="agree-disagree-button" disabled={this.state.buttonDisabled} onClick={(e) => this._handleClickDisagree(e)}><span className="agree-disagree-icon fa fa-times"></span></button>
					<AgreeDisagreeScore choice={"disagree_score"} clubOrPlayer={"player"}/>

				</div>

			</div>

			);

		return result;
	}
}


// ===============
// Connect redux state container to PlayerPage component
// ===============


const mapStateToProps = state => {
  return { 
  		playersViewed: state.playersViewed,
	  	agreeDisagreeScores: state.agreeDisagreeScores
  	};
};


export const AgreeDisagree = connect(mapStateToProps, { setPlayersAgreeDisagreeScore, locallyUpdateAgreeDisagreeScore })(AgreeDisagreeContainer);


