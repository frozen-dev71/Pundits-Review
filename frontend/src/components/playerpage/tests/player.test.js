// ======================================
// This file tests for the components listed in the PLAYERPAGE sub-directory
// ======================================


// ===========
// TEST COVERAGE
// ===========

// 1. Check that PlayerPage renders correctly
// 2. Check that AgreeDisagree renders correctly
// 3. Check that AgreeDisagreeScore renders correctly
// 4. Check that PlayerInfoContainer renders correctly
// 5. Check that ReviewScoreContainer renders correctly
// 6. Check that SeasonPerformanceContainer renders correctly
// 7. Check that SnippetItem renders correctly
// 8. Check that SnippetsContainer renders correctly
// 9. Check that SeasonPerformanceGraphWrapper renders correctly

// ===========
// IMPORTS 
// ===========


// General Imports
import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';

// Local JS Imports
import { PlayerPage } from '.././playerpage';
import { AgreeDisagree } from '.././components/agree_disagree';
import { AgreeDisagreeScore } from '.././components/agree_disagree_score';
import { PlayerInfoContainer } from '.././components/player_info_container';
import { ReviewScoreContainer } from '.././components/review_score_container';
import { SeasonPerformanceContainer } from '.././components/season_performance_container';
import { SnippetItem } from '.././components/snippet_item';
import { SnippetsContainer } from '.././components/snippets_container';
import { SeasonPerformanceGraphWrapper } from '.././components/season_performance/svg_container';

import store from '../../store/store';

// Configurations
Enzyme.configure({ adapter: new Adapter() })

// ===========
// Suppression of SSL Warning 
//
// Code adapted from comment from sketchbuch on GitHub issues post 'Warning: useLayoutEffect does nothing on the server #1373' on 12 Oct 2019
// https://github.com/reduxjs/react-redux/issues/1373
// Accessed 09 Sep 2020
//
// __Note__:
//	
//		- Jest produces "Warning: useLayoutEffect does nothing on the server" & 'Warning: React.createElement: type is invalid -- expected a string (for built-in components)' - warning which confuses test cases 
// 		- Wrapper test suppresses that warning for clear test reports
// ===========

describe('Player Page Component Tests', () => {
  	const originalConsoleError = console.error;

  	beforeEach(() => {
    	console.error = jest.fn((msg) => {
	      	if (msg.includes('Warning: useLayoutEffect does nothing on the server') || msg.includes('Warning: React.createElement: type is invalid -- expected a string (for built-in components)')) {
	       		return null;
	      	} else {
	        	originalConsoleError(msg);
	      	}
    	});
  	});

  afterEach(() => {
    console.error = originalConsoleError;
  });


// End of referenced code

  	// ===========
	// TEST 1. Check that PlayerPage renders correctly
	// ===========

	describe('PlayerPage', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><PlayerPage/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 2. Check that AgreeDisagree renders correctly
	// ===========

	describe('AgreeDisagree', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><AgreeDisagree/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 3. Check that AgreeDisagreeScore renders correctly
	// ===========

	describe('AgreeDisagreeScore', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><AgreeDisagreeScore/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 4. Check that PlayerInfoContainer renders correctly
	// ===========

	describe('PlayerInfoContainer', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><PlayerInfoContainer/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 5. Check that ReviewScoreContainer renders correctly
	// ===========

	describe('ReviewScoreContainer', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><ReviewScoreContainer/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 6. Check that SeasonPerformanceContainer renders correctly
	// ===========

	describe('SeasonPerformanceContainer', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><SeasonPerformanceContainer/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 7. Check that SnippetItem renders correctly
	// ===========

	describe('SnippetItem', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><SnippetItem/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 8. Check that SnippetsContainer renders correctly
	// ===========

	describe('SnippetsContainer', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><SnippetsContainer/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 9. Check that SeasonPerformanceGraphWrapper renders correctly
	// ===========

	describe('SeasonPerformanceGraphWrapper', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><SeasonPerformanceGraphWrapper/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


});
