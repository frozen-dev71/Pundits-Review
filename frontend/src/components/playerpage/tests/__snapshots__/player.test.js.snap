// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Page Component Tests AgreeDisagree should render correctly 1`] = `
<div
  class="agree-disagree-wrapper"
>
  <div
    class="agree-disagree-button-container"
  >
    <p
      class="agree-disagree-heading-text body-text-light"
    >
      AGREE
    </p>
    <button
      class="agree-disagree-button"
      id="agree-button"
    >
      <span
        class="agree-disagree-icon fa fa-check"
      />
    </button>
    <span />
  </div>
  <div
    class="agree-disagree-button-container"
  >
    <p
      class="agree-disagree-heading-text body-text-light"
    >
      DISAGREE
    </p>
    <button
      class="agree-disagree-button"
      id="disagree-button"
    >
      <span
        class="agree-disagree-icon fa fa-times"
      />
    </button>
    <span />
  </div>
</div>
`;

exports[`Player Page Component Tests AgreeDisagreeScore should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(AgreeDisagreeScoreWrapper) />
  </Provider>
</Router>
`;

exports[`Player Page Component Tests PlayerInfoContainer should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(PlayerInfo) />
  </Provider>
</Router>
`;

exports[`Player Page Component Tests PlayerPage should render correctly 1`] = `
<div
  class="empty-page-wrapper"
>
  <p
    class="no-results-text body-text-light"
  >
    It looks like you haven't selected a player 
  </p>
  <div
    class="dividing-line background-dark"
  />
  <p
    class="no-results-sub-text body-text-light"
  >
    Head back to the homepage to make a new search for a player
  </p>
  <a
    class="react-link empty-page-home-link body-text-light"
    href="/"
  >
    HOME
  </a>
</div>
`;

exports[`Player Page Component Tests ReviewScoreContainer should render correctly 1`] = `
<div
  class="review-score-container-wrapper"
>
  <p
    class="review-score-pundits-title heading-text"
  >
    Pundits Review
  </p>
  <div
    class="review-score-circle body-text-light white-text background-light"
  />
  <div
    class="review-score-positive-negative-total-grid"
  >
    <p
      class="review-score-positive-text body-text-light"
    >
      Positive Reviews
    </p>
    <p
      class="review-score-negative-text body-text-light"
    >
      Negative Reviews
    </p>
    <p
      class="review-score-total-text body-text-light"
    >
      Total Reviews
    </p>
    <p
      class="review-score-positive-number body-text-light"
    />
    <p
      class="review-score-negative-number body-text-light"
    />
    <p
      class="review-score-total-number body-text-light"
    />
  </div>
</div>
`;

exports[`Player Page Component Tests SeasonPerformanceContainer should render correctly 1`] = `
<div
  class="season-performance-wrapper"
>
  <p
    class="season-performance-title heading-text"
  >
    Season Performance
  </p>
  <div
    id="season-performance-graph-canvas"
  >
    <svg
      class="body-text-light"
      id="performance-graph-svg-container"
    />
  </div>
  <p
    class="graph-description-text body-text-light"
  >
    * Data collection for the 2019/20 season ran from JUL 13 to JUL 27
  </p>
</div>
`;

exports[`Player Page Component Tests SeasonPerformanceGraphWrapper should render correctly 1`] = `
<svg
  class="body-text-light"
  id="performance-graph-svg-container"
/>
`;

exports[`Player Page Component Tests SnippetItem should render correctly 1`] = `
<div
  class="snippet-item-wrapper background-light"
>
  <div
    class="snippet-item-header-container background-dark"
  >
    <p
      class="snippet-item-media-source body-text-light white-text"
    />
    <p
      class="snippet-item-published-date body-text-light white-text"
    />
  </div>
  <p
    class="snippet-item-article-phrase body-text-light"
  >
    <span
      class="snippet-item-quote-icon"
    >
      “
    </span>
    <span
      class="snippet-item-quote-icon"
    >
      ”
    </span>
  </p>
  <div
    class="snippet-item-bottom-container"
  >
    <a
      class="react-link"
      target="_blank"
    >
      <button
        class="snippet-item-read-article-button body-text-light"
      >
        READ THE ARTICLE
      </button>
    </a>
  </div>
</div>
`;

exports[`Player Page Component Tests SnippetsContainer should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <SnippetsContainer />
  </Provider>
</Router>
`;
